corehq/apps/app_manager/xform.py

```diff
@@ -496,32 +497,42 @@ def update_block(self):
         return update_block
 
     def add_case_updates(self, updates, make_relative=False):
+        from corehq.apps.app_manager.models import ConditionalCaseUpdate
         update_block = self.update_block
         if not updates:
             return
 
         update_mapping = {}
         attachments = {}
         for key, value in updates.items():
-            value = getattr(value, "question_path", value)
+            value_str = getattr(value, "question_path", value)
             if key == 'name':
                 key = 'case_name'
-            if self.is_attachment(value):
-                attachments[key] = value
+            if self.is_attachment(value_str):
+                attachments[key] = value_str
             else:
                 update_mapping[key] = value
 
         for key, q_path in sorted(update_mapping.items()):
+            resolved_path = self.xform.resolve_path(q_path)
+            edit_mode_path = ''
+            if isinstance(q_path, ConditionalCaseUpdate) and q_path.update_mode == UPDATE_MODE_EDIT:
+                if 'commcare_usercase' in self.path:
```

Comment:
if we're using the path to get the case ID then we don't need this if statement since it will work the same

Reply:
The same approach will work for any case type since it's directly referencing the case ID in the case block itself.

Reply:
@stephherbers this might need follow-up too (different PR)? You may have added a change already too, I'm not sure.

Reply:
I added the change here: [#30910](https://github.com/dimagi/commcare-hq/pull/30910)
thanks for clarifying, Simon that was another gap in my understanding